File: CoreStore/CoreStore/CoreStore.Web/Modules/Store/Categories/CategoriesForm.cs

namespace CoreStore.Store.Forms
{
    using Serenity.ComponentModel;
    using System;

    [FormScript("Store.Categories")]
    [BasedOnRow(typeof(Entities.CategoriesRow), CheckNames = true)]
    public class CategoriesForm
    {
        public Int32 CategoryTypeId { get; set; }
        public String CategoryCode { get; set; }
        public String CategoryName { get; set; }
        public String Description { get; set; }
    }
}

File: Ser2023/Store/Store.Web/Modules/ServerTypes/Store/CategoriesForm.ts
import { CategoriesTypeDialog } from "@/Store/CategoriesType/CategoriesTypeDialog";
import { LookupEditor, StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CategoriesForm {
    CategoryTypeId: LookupEditor;
    CategoryCode: LookupEditor;
    CategoryName: StringEditor;
    Description: StringEditor;
    Picture: StringEditor;
}

export class CategoriesForm extends PrefixedContext {
    static formKey = 'Store.Categories';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoriesForm.init)  {
            CategoriesForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;

            initFormType(CategoriesForm, [
                'CategoryTypeId', w0,
                'CategoryCode', w0,
                'CategoryName', w1,
                'Description', w1,
                'Picture', w1
            ]);
        }
    }
}

[CategoriesTypeDialog]; // referenced types
